iothub/device/src/Twin/DesiredProperties.cs

```diff
@@ -23,7 +23,7 @@ public class DesiredProperties : PropertyCollection
         /// This class can be inherited from and set by unit tests for mocking purposes.
         /// </remarks>
         protected internal DesiredProperties(Dictionary<string, object> desiredProperties)
-            : base(desiredProperties, true)
+            : base(desiredProperties, false)
```

Comment:
If this is a mocking ctor, false makes more sense here to just add the properties to the internal dictionary